src/main/java/de/ace/html2pdf/application/PdfService.java

```diff
@@ -1,29 +1,91 @@
 package de.ace.html2pdf.application;
 
-import de.ace.html2pdf.config.DavidPDFException;
-import de.ace.html2pdf.model.PdfRequest;
+
+import de.ace.html2pdf.model.FooterImageProperties;
+import de.ace.html2pdf.model.FooterProperties;
+import de.ace.html2pdf.model.PdfData;
 import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.PDPageContentStream;
+import org.apache.pdfbox.pdmodel.graphics.image.PDImageXObject;
+import org.apache.pdfbox.rendering.PDFRenderer;
 import org.springframework.stereotype.Service;
 
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
 import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import static de.ace.html2pdf.config.Constants.IMAGE_HEIGHT;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
 public class PdfService {
 
     private final PdfRenderComponent pdfRenderComponent;
-    private final PostProcessorPdfComponent postProcessorPdfComponent;
 
-    public byte[] convert(PdfRequest pdfRequest) {
-        try (ByteArrayOutputStream bs = new ByteArrayOutputStream()) {
+    public byte[] html2pdf(String html) throws IOException {
```

Comment:
This method throws IOException, but is not caught in the controller. We can either
 i) catch this exception in the service layer itself  OR 
 ii) propagate all the exceptions to Controller layer and catch it in the controller layer OR
  iii) propagate it to the GlobalExceptionHandler and let it handle the exception (we need an additional handler method)
  
  I am not sure which is the best approach here, may be @KirkDerAuslaender can comment more on it.